crates/matrix/src/client/resources/error.rs

```diff
@@ -0,0 +1,5 @@
+#[derive(Clone, Debug, Deserialize)]
+pub struct MatrixError {
+    pub errcode: String,
+    pub error: String,
```

Comment:
You can use sere to rename on deserialization. That way you could write error code correctly.

```suggestion
    pub error_code: String,
    pub error: String,
```

Reply:
I suggest using enum for Error code, otherwise we still have to check for arbitrary strings in consumer end

Reply:
Was planning to but seems like I overlooked it, thanks for the reminder!